server/src/inmantals/server.py

```diff
@@ -109,11 +122,33 @@ def treeify_reverse(iterator):
                 for k, v in groupby(anchormap, lambda x: x[1].file)
             }
 
+            await self.publish_diagnostics(None)
+
+        except CompilerException as e:
+            if ast_export is not None:
+                error: ast_export.Error = e.export()
+                params: Optional[lsp_types.PublishDiagnosticsParams]
+                if error.location is None:
+                    params = None
+                else:
+                    params = lsp_types.PublishDiagnosticsParams(
+                        uri=error.location.uri,
+                        diagnostics=[
+                            lsp_types.Diagnostic(
+                                range=error.location.range,
+                                severity=lsp_types.DiagnosticSeverity.Error,
+                                message=f"{error.type}: {error.message}",
+                            )
+                        ],
+                    )
+                await self.publish_diagnostics(params)
+            logger.exception("Compile failed")
+
         except Exception:
             logger.exception("Compile failed")
 
     async def initialized(self):
-        await asyncio.get_event_loop().run_in_executor(self.threadpool, self.compile_and_anchor)
+        await self.compile_and_anchor()
```

Comment:
@wouterdb I'm not sure about this change, wdyt?

Reply:
1. I'm not sure this is according to the language server spec
2. Why block the IDE side on this?

Reply:
Does this really block the IDE side? Cause this is all happening in the language server, right?
Anyway, the reason I made this change is because `compile_and_anchor` now does an async call to `send_notification`. I thought this might not be the right approach to deal with that but I'm not sure what would be. Do you have a suggestion?

Reply:
https://docs.python.org/3/library/asyncio-task.html#creating-tasks

Reply:
Sorry, I'm wrong, this logically blocking already.

The reason for the threadpool is because compilation can take some time, blocking the IOLoop, making the whole lsp unresponsive. so you can move the executor call deeper into the structure

Reply:
I've added a comment above the `run_in_executor` line. Could you verify that it describes the motivation for using the executor correctly?

Reply:
I think it is good